src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -147,12 +159,53 @@ public boolean removePerson(Person key) throws PersonNotFoundException {
         }
     }
 
+    public void sortByName() {
+        persons.sortPersons();
+    }
+
     //// tag-level operations
 
     public void addTag(Tag t) throws UniqueTagList.DuplicateTagException {
         tags.add(t);
     }
 
+    /**
+     * Calls removeTagFromPerson method when tag is found in tags.
+     * @param t
+     * @throws TagNotFoundException
+     */
+    public void removeTag(Tag t) throws TagNotFoundException {
+        if (tags.contains(t)) {
+            for (Person person : persons) {
+                removeTagFromPerson(t, person);
+            }
+            tags.remove(t);
+        } else {
+            throw new TagNotFoundException("Specific tag is not used in the address book.");
+        }
+    }
+
+    /**
+     * Removes a specific tag from an individual person and updates the person's information.
+     * @param tag
+     * @param person
```

Comment:
Description for the parameters are needed, although it is obvious. This is because other developers may not know the link between the tag and person here